synapse/storage/databases/main/cache.py

```diff
@@ -98,6 +114,17 @@ def __init__(
         else:
             self._cache_id_gen = None
 
+        if hs.config.worker.run_background_tasks and isinstance(
```

Comment:
Moving this here as I first looked for the context on why this is Postgres only, only to find it in the next lines which I didn't read to before first scanning the rest of the PR.

```suggestion
        # Remove old rows from the `cache_invalidation_stream_by_instance` table
        # automatically (this table is unused in SQLite). This is only applicable
        # if we are on Postgres and therefore populate those tables.
        if hs.config.worker.run_background_tasks and isinstance(
```

Was also trying to think of better phrasing but didn't come up with something concretely better. Up to you

```
            # Since we only populate these tables when using Postgres, we only
            # need to clean up when using Postgres
```

```
            # We only ever populate these tables when using Postgres, so it's only
            # applicable to do the clean up when using Postgres
```